### PR TITLE
Serial recovery storage erase cmd

### PR DESCRIPTION
**Pull request by de-nordic**
PR contains tow commits:
 - adding support for system specific mcumgr commands in serial recovery;
 - adding zephyr implementation of the command that erases storage partition.

### CODE REVIEW

**Comment by de-nordic** [comment]
FYI @sigvartmh

**Review by d3zd3z — APPROVED** [review]

File: boot/boot_serial/src/boot_serial.c [hunk 1]
```diff
@@ -58,6 +58,12 @@
 #include "bootutil_priv.h"
 #endif
 
+#ifdef CONFIG_ENABLE_MGMT_PERUSER
```
Comment (nvlsianpu) [inline boot/boot_serial/src/boot_serial.c:61]: I would use mediation of mcuboot_config.h file

Comment (de-nordic) [reply boot/boot_serial/src/boot_serial.c:61]: OK, will do.

**Review by nvlsianpu — APPROVED** [review]

**Comment by nvlsianpu** [comment]
@utzig If no obstacles I would merge that.

**Comment by utzig** [comment]
> @utzig If no obstacles I would merge that.

Sure, don't wait for me, I am totally starved from time to review stuff here.

**Pull request merged**